### PR TITLE
Fees-register-frontend non prod kustmization

### PR DESCRIPTION
**Pull request by AnilDhawaleGitHub**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X ] No
```

### CODE REVIEW

File: k8s/namespaces/fees-pay/fees-register-frontend/ithc.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+apiVersion: helm.fluxcd.io/v1
+kind: HelmRelease
+metadata:
+  name: fees-register-frontend
+spec:
+  releaseName: fees-register-frontend
+  values:
+    nodejs:
+      image: hmctspublic.azurecr.io/fees-register/frontend:prod-8e30a3d-20220214154643 #{"$imagepolicy": "flux-system:fees-register-frontend"}
+
```
Comment (thomast1906) [inline k8s/namespaces/fees-pay/fees-register-frontend/ithc.yaml:10]: Please resolve spacing

File: k8s/namespaces/fees-pay/fees-register-frontend/demo.yaml [hunk 2]
```diff
@@ -3,28 +3,14 @@ apiVersion: helm.fluxcd.io/v1
...
-    retry: true
-  chart:
-    git: git@github.com:hmcts/hmcts-charts
-    ref: master
-    path: stable/fees-register-frontend
   values:
     nodejs:
-      replicas: 2
-      useInterpodAntiAffinity: true
       image: hmctspublic.azurecr.io/fees-register/frontend:pr-401
```
Comment (thomast1906) [inline k8s/namespaces/fees-pay/fees-register-frontend/demo.yaml:13]: Missing image policy

File: k8s/namespaces/fees-pay/fees-register-frontend/demo.yaml [hunk 3]
```diff
@@ -3,28 +3,14 @@ apiVersion: helm.fluxcd.io/v1
 kind: HelmRelease
 metadata:
   name: fees-register-frontend
-  namespace: fees-pay
   annotations:
     flux.weave.works/automated: "true"
```
Comment (thomast1906) [inline k8s/namespaces/fees-pay/fees-register-frontend/demo.yaml:7]: Is this needed?

File: k8s/namespaces/fees-pay/fees-register-frontend/ithc.yaml (hunk 1 above)
Comment (AnilDhawaleGitHub) [reply k8s/namespaces/fees-pay/fees-register-frontend/ithc.yaml]: its done

File: k8s/namespaces/fees-pay/fees-register-frontend/demo.yaml (hunk 3 above)
Comment (AnilDhawaleGitHub) [reply k8s/namespaces/fees-pay/fees-register-frontend/demo.yaml:6]: yes i have removed the **namespace: fees-pay** as its present in here
spec:
  releaseName: fees-register-frontend

Comment (thomast1906) [reply k8s/namespaces/fees-pay/fees-register-frontend/demo.yaml:6]: I dont think below is needed?

```
flux.weave.works/automated: "true"
 flux.weave.works/tag.nodejs: pr-401-*
```

File: k8s/namespaces/fees-pay/fees-register-frontend/ithc.yaml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+apiVersion: helm.fluxcd.io/v1
+kind: HelmRelease
+metadata:
+  name: fees-register-frontend
```
Comment (thomast1906) [inline k8s/namespaces/fees-pay/fees-register-frontend/ithc.yaml:4]: File doesn't look to be needed

File: k8s/namespaces/fees-pay/fees-register-frontend/aat.yaml [hunk 5]
```diff
@@ -0,0 +1,11 @@
+apiVersion: helm.fluxcd.io/v1
+kind: HelmRelease
+metadata:
+  name: fees-register-frontend
+spec:
+  releaseName: fees-register-frontend
+  values:
+    nodejs:
+      image: hmctspublic.azurecr.io/fees-register/frontend:prod-8e30a3d-20220214154643 #{"$imagepolicy": "flux-system:fees-register-frontend"}
```
Comment (thomast1906) [inline k8s/namespaces/fees-pay/fees-register-frontend/aat.yaml:9]: image looks to be duplicated of fees-register-frontend.yaml? 
Please only include specific environment differences in aat.yaml etc

Example here: https://github.com/hmcts/cnp-flux-config/blob/master/k8s/namespaces/bar/bar-api/aat.yaml

File: k8s/namespaces/fees-pay/fees-register-frontend/demo.yaml [hunk 6]
```diff
@@ -0,0 +1,12 @@
+apiVersion: helm.fluxcd.io/v1
+kind: HelmRelease
+metadata:
+  name: fees-register-frontend
+spec:
+  releaseName: fees-register-frontend
+  values:
+    nodejs:
+      image: hmctspublic.azurecr.io/fees-register/frontend:pr-401 #{"$imagepolicy": "flux-system:demo-fees-register-frontend"}
```
Comment (thomast1906) [inline k8s/namespaces/fees-pay/fees-register-frontend/demo.yaml:9]: imagetag pr-401 does not look correct, image policy will update correctly. 

I am assuming pr-401 is not the specific tag you require for demo?

File: k8s/namespaces/fees-pay/fees-register-frontend/ithc.yaml [hunk 7]
```diff
@@ -0,0 +1,9 @@
...
+spec:
+  releaseName: fees-register-frontend
+  values:
```
Comment (thomast1906) [inline k8s/namespaces/fees-pay/fees-register-frontend/ithc.yaml:7]: I see in your removal above
```
    global:
      environment: ithc
      tenantId: "531ff96d-0ae9-462a-8d2d-bec7c0b42082"
      enableKeyVaults: true
```

Are these going to be needed for ithc?

File: k8s/namespaces/fees-pay/fees-register-frontend/perftest.yaml [hunk 8]
```diff
@@ -3,25 +3,11 @@ apiVersion: helm.fluxcd.io/v1
...
   values:
     nodejs:
-      replicas: 2
-      useInterpodAntiAffinity: true
       image: hmctspublic.azurecr.io/fees-register/frontend:perftest #{"$imagepolicy": "flux-system:perftest-fees-register-frontend"}
       imagePullPolicy: Always
       environment:
         DUMMY_RESTART_VAR: true
-    global:
-      environment: perftest
```
Comment (thomast1906) [inline k8s/namespaces/fees-pay/fees-register-frontend/perftest.yaml:25]: are these global values needed for perftest?

File: k8s/namespaces/fees-pay/fees-register-frontend/demo.yaml (hunk 6 above)
Comment (AnilDhawaleGitHub) [reply k8s/namespaces/fees-pay/fees-register-frontend/demo.yaml:9]: updated it to pr no 432 :: same is there  we are pointing in apps/fees-pay/fees-register-frontend/demo-image-policy.yaml

File: k8s/namespaces/fees-pay/fees-register-frontend/ithc.yaml [hunk 9]
```diff
@@ -0,0 +1,9 @@
...
+    nodejs:
+      image: hmctspublic.azurecr.io/fees-register/frontend:prod-8e30a3d-20220214154643 #{"$imagepolicy": "flux-system:fees-register-frontend"}
```
Comment (thomast1906) [inline k8s/namespaces/fees-pay/fees-register-frontend/ithc.yaml:9]: Can you remove?

File: k8s/namespaces/fees-pay/fees-register-frontend/ithc.yaml (hunk 7 above)
Comment (thomast1906) [reply k8s/namespaces/fees-pay/fees-register-frontend/ithc.yaml:7]: as discussed, not needed

File: k8s/namespaces/fees-pay/fees-register-frontend/ithc.yaml (hunk 9 above)
Comment (AnilDhawaleGitHub) [reply k8s/namespaces/fees-pay/fees-register-frontend/ithc.yaml]: its removed

File: k8s/namespaces/fees-pay/fees-register-frontend/perftest.yaml [hunk 10]
```diff
@@ -3,25 +3,11 @@ apiVersion: helm.fluxcd.io/v1
...
-  chart:
-    git: git@github.com:hmcts/hmcts-charts
-    ref: master
-    path: stable/fees-register-frontend
   values:
     nodejs:
-      replicas: 2
-      useInterpodAntiAffinity: true
       image: hmctspublic.azurecr.io/fees-register/frontend:perftest #{"$imagepolicy": "flux-system:perftest-fees-register-frontend"}
       imagePullPolicy: Always
```
Comment (thomast1906) [inline k8s/namespaces/fees-pay/fees-register-frontend/perftest.yaml:11]: is this needed?

File: k8s/namespaces/fees-pay/fees-register-frontend/demo.yaml [hunk 11]
```diff
@@ -0,0 +1,12 @@
+apiVersion: helm.fluxcd.io/v1
+kind: HelmRelease
+metadata:
+  name: fees-register-frontend
+spec:
+  releaseName: fees-register-frontend
+  values:
+    nodejs:
+      image: hmctspublic.azurecr.io/fees-register/frontend:pr-432 #{"$imagepolicy": "flux-system:demo-fees-register-frontend"}
+      imagePullPolicy: Always
```
Comment (thomast1906) [inline k8s/namespaces/fees-pay/fees-register-frontend/demo.yaml:10]: not needed

File: k8s/namespaces/fees-pay/fees-register-frontend/ithc.yaml [hunk 12]
```diff
@@ -0,0 +1,6 @@
+apiVersion: helm.fluxcd.io/v1
+kind: HelmRelease
+metadata:
+  name: fees-register-frontend
+spec:
+  releaseName: fees-register-frontend
```
Comment (thomast1906) [inline k8s/namespaces/fees-pay/fees-register-frontend/ithc.yaml:6]: Is ithc.yaml missing?

```
      environment:
        DUMMY_RESTART_VAR: true
```

Comment (AnilDhawaleGitHub) [reply k8s/namespaces/fees-pay/fees-register-frontend/ithc.yaml:6]: added environment:
        DUMMY_RESTART_VAR: true

File: k8s/namespaces/fees-pay/fees-register-frontend/perftest.yaml (hunk 10 above)
Comment (AnilDhawaleGitHub) [reply k8s/namespaces/fees-pay/fees-register-frontend/perftest.yaml:11]: as this is common so i kept it in fees-register-frontend.yaml  and removed from env specific files 
       replicas: 2
      useInterpodAntiAffinity: true

**Review by thomast1906 — APPROVED** [review]

**Review by arbab-cog — APPROVED** [review]

File: k8s/namespaces/fees-pay/fees-register-frontend/demo.yaml (hunk 11 above)
Comment (AnilDhawaleGitHub) [reply k8s/namespaces/fees-pay/fees-register-frontend/demo.yaml]: its done

**Pull request merged**